nevergrad/functions/mixsimulator/core.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Based on https://github.com/Foloso/MixSimulator/tree/nevergrad_experiment
+
+import numpy as np
+from ..base import ExperimentFunction
+from mixsimulator.MixSimulator import MixSimulator  #type: ignore
+
+
+class OptimizeMix(ExperimentFunction):
+    """
+        MixSimulator is an application with an optimization model for calculating 
+        and simulating the least cost of an energy mix under certain constraints.
+        
+        For now, it uses a default dataset (more will be added soon).
+        
+        For more information, visit : https://github.com/Foloso/MixSimulator     
+        
+        Parameters
+        ----------
+        time: int
+            total time over which it evaluates the mix (must in hour)
+                
+    """
+    def __init__(self,time: int = 168) -> None:
+        
+        parameters = self._create_parametrization(time = time)
+        super().__init__(self._simulate_mix , parameters)
+        
+        self.register_initialization(time=time)
+        self.add_descriptors(time=time)
+
+    def _simulate_mix(self, x: np.ndarray) -> float:
+        mix = MixSimulator()
+        mix.set_data_to("Toamasina")
+        mix.set_penalisation_cost(100)
+        mix.set_carbon_cost(10)
```

Comment:
it feels weird to have to recreate this each time, is there a reason? Wouldn't that be better to instantiate it in the init and keep it as an attribute? This way `_create_parametrization` will not be needed anymore either, since everything could be done from the init.

Also, it's better to avoid this code duplication between `simulate_mix` and `create_parametrization`, because if eventually something is changed (eg: Toamasina), then no need to change it twice.
Finally, it looks like even _simulate_mix could be removed, if you then use `super().__init__(self._mix.loss_function, parameters)`
Less code is always better :D

Reply:
You are right :D
Our first goal was to make everything work. But we already work on it to make it better now.

Reply:
Yes thank you for your help :-) and sorry for the delay. We work on it. Just wanted to make it work first :-)

Reply:
no worries in term of delays, I am not always fast to answer, sorry :s